src/core/service.c

```diff
@@ -3549,14 +3553,42 @@ static int service_dispatch_exec_io(sd_event_source *source, int fd, uint32_t ev
                                 log_unit_debug(UNIT(s), "Got EOF on exec-fd while it was disabled, ignoring.");
 
                         return 0;
+
+                case sizeof(uint8_t):
+                        /* A single byte was read, which means the exec fd logic is now off (unless somehow
+                         * an old executor has written back to a new pid1, so also allow x=1 just in case).
+                         * Also reset the exec timestamp, given the execve actually failed. */
+
+                        s->exec_fd_hot = x;
+                        if (s->state == SERVICE_START)
+                                s->main_exec_status.handover_timestamp = DUAL_TIMESTAMP_NULL;
+
+                        continue;
+
+                case sizeof(uint64_t): {
```

Comment:
{} not necessary.

Reply:
We define new variable here, so {} is necessary according to our coding style

Reply:
For some reason I thought it was necessary when declaring new variables inside a case block